### PR TITLE
Add WYSIWYG control type

### LINKED ISSUE #50: Should there be WYSIWYG Control/Fields?
Seems like plugin devs might expect to be able to quickly add a WYSIWYG/TinyMCE editor to the backend using the Fields API. I don’t see a way to do so now. Apologies if I missed anything about it in the Issues or docs. But if it hasn’t already been discussed, are there any chances you’d be willing to add it? I think it makes sense in core instead of being left to 3rd parties.

### PR DESCRIPTION
**Pull request by sc0ttkclark**
Fixes #50

### CODE REVIEW

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @package    WordPress
+ * @subpackage Fields_API
+ */
+
+/**
+ * Fields API Textarea Control class.
```
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: Small text tweak here from Textarea >> WYSIWYG

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	 * {@inheritdoc}
+	 */
+	public $type = 'wysiwyg';
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function render_content() {
+		
+		$this->input_attrs();
```
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: Remove this

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		wp_editor( esc_html( $this->value() ), $this->input_attrs['id'] );
```
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: Get `$input_attrs` via `$input_attrs = $this->get_input_attrs();`.

Add $settings to third parameter, pass in `'textarea_name' => $input_attrs['name']`

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		echo '<pre>', var_dump( $this->input_attrs ), "\n", $this->id, "\n", $this->value, '</pre>';
+		spew_var($this);
```
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: Remove this

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	public $type = 'wysiwyg';
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function render_content() {
+		
+		$this->input_attrs();
+		wp_editor( esc_html( $this->value() ), $this->input_attrs['id'] );
+		echo '<pre>', var_dump( $this->input_attrs ), "\n", $this->id, "\n", $this->value, '</pre>';
```
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: Remove this

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+		spew_var($this);
+
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function render_attrs() {
```
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: You can remove this now (pull latest from fields api `develop` branch over to yours to get the changes.

File: implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php (hunk 3 above)
Comment (sc0ttkclark) [inline implementation/wp-includes/fields-api/control-types/class-wp-fields-api-wysiwyg-control.php]: Remove esc_html(), value gets escaped through wp_editor()

**Comment by ymazuz** [comment]
Ok, I think the commits I added to my repo are just in this PR now. So it looks basically right on my end.

**Comment by paulwilde** [comment]
It would be nice if this field would also allow to specify the settings which can be provided into `wp_editor()`'s third parameter rather than being hard-coded with only the `textarea_name`.

**Comment by sc0ttkclark** [comment]
@paulwilde that's next :)

**Comment by ymazuz** [comment]
I’ll take a crack at that today, assuming you haven’t already started.

**Comment by sc0ttkclark** [comment]
Already did in https://github.com/sc0ttkclark/wordpress-fields-api/commit/e1f5370aa0ee086ba30d2e2b302b3f83a84fa53d :)

**Comment by ymazuz** [comment]
Pulled your latest and noticed that right before you commented. From that commit it looks like the `$this->editor_settings` is always empty, but if you’re in the middle of this I’ll leave it alone!

**Comment by sc0ttkclark** [comment]
It's empty by default right now, but you can pass in options for it into the control arguments directly with 'editor_settings' option.

**Pull request merged**